### PR TITLE
Enumerable methods ruby

### PR DESCRIPTION
**Pull request by DanteAlonsoHT**
**Hello everybody!**

For the next project, I've been working on re-creating the enumerable methods using my own style from scratch using Ruby. in this pull request I've been working on the next files:

- _enumerable_methods_ruby.rb_
- _README.md_

I've passed the linters successfully. 👍

### CODE REVIEW

File: enumerable_methods_ruby.rb [hunk 1]
```diff
@@ -1 +1,132 @@
...
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return false if i == param }
+    elsif block_given? && param.instance_of?(Range)
+      to_a.my_each { |i| return false if yield i }
+      return true
+    else
+      to_a.my_each { |i| return false if yield i }
+    end
+    true
+  end
```
Comment (george-shammar) [inline enumerable_methods_ruby.rb:84]: - [ ] my_none? When no block or argument is given, it should return false only if one of the collection members is true.

File: enumerable_methods_ruby.rb [hunk 2]
```diff
@@ -1 +1,132 @@
...
+      to_a.my_each { |i| return true if [i.class, i.class.superclass].include?(param) }
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return true if i == param }
+    elsif param.is_a?(Regexp)
+      to_a.my_each { |i| return true if i.match(param) }
+    else
+      to_a.my_each { |i| return true if yield i }
+    end
+    false
+  end
```
Comment (george-shammar) [inline enumerable_methods_ruby.rb:63]: - [ ] my_any? When a Regex is passed as an argument, it should return true if any of the collection matches the Regex.

File: enumerable_methods_ruby.rb [hunk 3]
```diff
@@ -1 +1,132 @@
...
+      to_a.my_each { |i| return false unless [i.class, i.class.superclass].include?(param) }
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return false unless i == param }
+    elsif !block_given? && (param.is_a? Regexp)
+      to_a.my_each { |i| return false unless i.match(param) }
+    elsif block_given?
+      to_a.my_each { |i| return false unless yield i }
+    end
+    true
+  end
```
Comment (george-shammar) [inline enumerable_methods_ruby.rb:47]: - [ ]  my_all?, when a Regex is passed as an argument, it should return true if all of the collection matches the Regex.

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED** :recycle: :recycle: 

This is absolutely a very strong first submission. :+1: :clap:  

You should be really proud. You have very minimal improvements to implement. Great job. :+1: 

Please see my inline comments for specific changes required to get this project approved.

Please submit another CRR when you're ready.

Happy coding. :+1: :+1:

**Review by henatan99 — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 

Hi there, wonderful work you have done on this project :clap: :clap: :clap: . You are a single bug fix away to have your project approved.

## To Highlight 
- Your enumerable methods work as the standard methods except for one bug

## To Improve 
- enumerable #my_none? when no block or argument is given should return true only if none of the collection members is true
- Please see the example below which compares the result of your method and the standard none? method.
:point_right: array = [nil]
:point_right: array.my_none?  => false 
:point_right: array.none? => true 

Same is true when array = [nil, false]

### Kindly fix this bug and submit it for another review 

If you have any question, you can reach out to me via 
[LinkedIn](https://www.linkedin.com/in/henok-mekonnen/) | [Twitter](https://twitter.com/henatan99) | Henok Mekonnen on Slack 

## Happy Coding :computer:

File: enumerable_methods_ruby.rb [hunk 4]
```diff
@@ -1 +1,132 @@
...
+    elsif !block_given? && !param.nil?
+      to_a.my_each { |i| return false if i == param }
+    elsif block_given? && param.instance_of?(Range)
+      to_a.my_each { |i| return false if yield i }
+      return true
+    else
+      to_a.my_each { |i| return false if yield i }
+    end
+    true
+  end
```
Comment (Berabjesus) [inline enumerable_methods_ruby.rb]: - [x] when no block or argument is given returns false only if one of the collection members is true.

```
  p [nil, false, true, []].my_none? # true
  p [nil, false, true, []].none? # false
```

**Review by Berabjesus — CHANGES_REQUESTED** [review]
# Status: CHANGES REQUIRED :warning: :construction: 
Hi @DanteAlonsoHT great job implementing the requested changes so far. 👏 

### However, based on the previous reviews, you still got to fix your `my_none` enum method to proceed

## Mandatory
#### `my_none`
- [x]  when no block or argument is given returns false only if one of the collection members is true (check inline comments)

## Optional
- [x] ❗ Your project should have followed Git Flow. Meaning that you should have created a `development` branch then make it the default branch. Then create your `feature` branch from the `development` branch. Please make sure you follow Git flow whenever it's required in your project specification.
- [x]  For `my_any?` when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [x] for `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex

## **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

File: enumerable_methods_ruby.rb (hunk 4 above)
Comment (DanteAlonsoHT) [reply enumerable_methods_ruby.rb]: image

I've seen it on my local repository, but in my project, it works properly, I don't know why in your case it doesn't work :c

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @DanteAlonsoHT, good job on your enumerables!! 🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

shine team


Now you can merge this branch 🚀 

You can contact me on slack if you have any questions, just search for me as @ Miguel Ricaño.

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

**Pull request merged**